dns-rc-autoscaler/nanny/nanny_lib.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package nanny implements logic to poll the k8s apiserver for cluster status,
+and update a deployment based on that status.
+*/
+package nanny
+
+import (
+	"log"
+	"time"
+)
+
+// KubernetesClient is an object that performs the nanny's requisite interactions with Kubernetes.
+type KubernetesClient interface {
+	CountNodes() (uint64, uint64, error)
+	PodReplicas() (uint64, error)
+	UpdateReplicas(uint64) error
+	GetParentRc(namespace, podname string) (string, error)
+}
+
+// PollAPIServer periodically counts the number of nodes, estimates the expected
+// number of replicas, compares them to the actual replicas, and
+// updates the parent ReplicationController with the expected replicas if necessary.
+func PollAPIServer(k8s KubernetesClient, scaler Scaler, pollPeriod time.Duration, verbose bool) {
```

Comment:
I'm going to guess he polls because the addon-resizer polls. KISS was the recommendation from the API guys, but the client has improved a lot since then.